eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Deleg.hs

```diff
@@ -255,34 +256,30 @@ delegationTransition = do
   TRC (DelegEnv slot ptr acnt pp, ds, c) <- judgmentContext
   case c of
     DCertDeleg (RegKey hk) -> do
-      eval (hk ∉ dom (_rewards ds)) ?! StakeKeyAlreadyRegisteredDELEG hk
-
+      eval (hk ∉ dom (rewards ds)) ?! StakeKeyAlreadyRegisteredDELEG hk
       pure $
-        ds
-          { _rewards = eval (_rewards ds ∪ singleton hk mempty),
-            _ptrs = eval (_ptrs ds ∪ singleton ptr hk)
-          }
+        ( let u1 = _unified ds
+              u2 = (Rewards u1 UM.∪ (hk, mempty))
+              u3 = (Ptrs u2 UM.∪ (ptr, hk))
+           in ds {_unified = u3}
+        )
     DCertDeleg (DeRegKey hk) -> do
       -- note that pattern match is used instead of cwitness, as in the spec
-      eval (hk ∈ dom (_rewards ds)) ?! StakeKeyNotRegisteredDELEG hk
-
-      let rewardCoin = Map.lookup hk (_rewards ds)
+      eval (hk ∈ dom (rewards ds)) ?! StakeKeyNotRegisteredDELEG hk
+      let rewardCoin = UM.lookup hk (rewards ds)
       rewardCoin == Just mempty ?! StakeKeyNonZeroAccountBalanceDELEG rewardCoin
 
       pure $
-        ds
-          { _rewards = eval (setSingleton hk ⋪ _rewards ds),
-            _delegations = eval (setSingleton hk ⋪ _delegations ds),
-            _ptrs = eval (_ptrs ds ⋫ setSingleton hk)
-          }
+        let u0 = _unified ds
+            u1 = (Set.singleton hk UM.⋪ Rewards u0)
+            u2 = (Set.singleton hk UM.⋪ Delegations u1)
+            u3 = (Ptrs u2 UM.⋫ Set.singleton hk)
+         in ds {_unified = u3}
```

Comment:
is there not a way to just remove `hk` from the unified map?

Reply:
We could combine these into one operation, but then we don't have the nice continuation between the old code, the spec,  and the new code.